### PR TITLE
chore: added skeleton loader

### PR DESCRIPTION
**Pull request by jey-exp**
Successful merge of this PR will solve #16

### CODE REVIEW

File: src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      <div className="space-y-4 pb-6 pt-7">
+        <section className="px-2 md:px-3">
+          <div className="space-y-1">
+            <Skeleton className="h-6 w-64" />
+            <Skeleton className="h-4 sm:w-96 w-64" />
+          </div>
+        </section>
+        <section className="px-0 md:px-3">
+          {Array.from(Array(4).keys()).map((index) => (
+            <GridContainer>
```
Comment (realChakrawarti) [inline src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx:15]: [Blocking] `index` needs be used as a key, `<GridContainer key={index}>`

File: src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+              {Array.from(
+                Array(4)
+                  .keys()
+                  .map((index) => <YouTubeSkeletonCard key={index} />)
+              )}
```
Comment (realChakrawarti) [inline src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx:20]: [Blocking] Throws error, misplaced parenthesis, apply suggested change

image

```suggestion
              {Array.from(
                Array(4)
                  .keys())
                  .map((index) => <YouTubeSkeletonCard key={index} />)
              }
```

File: src/app/(views)/(public)/a/[archiveId]/loading.tsx [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import Spinner from "~/components/custom/spinner";
```
Comment (realChakrawarti) [inline src/app/(views)/(public)/a/[archiveId]/loading.tsx:1]: [Blocking] Remove `Spinner` import as this is no longer required

**Review by realChakrawarti — CHANGES_REQUESTED** [review]
@jey-exp Thanks for updating the code. 👍🏻

Left a few [Blocking] comments, please address those. Once fixed, let me know, then its good to merge. 😊

File: src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx (hunk 1 above)
Comment (jey-exp) [reply src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx:15]: Thanks for pointing it out. Will fix it.

File: src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx (hunk 2 above)
Comment (jey-exp) [reply src/app/(views)/(public)/a/[archiveId]/skeleton-loader.tsx:20]: Sorry for the typo, will get it fixed.

File: src/app/(views)/(public)/a/[archiveId]/loading.tsx (hunk 3 above)
Comment (jey-exp) [reply src/app/(views)/(public)/a/[archiveId]/loading.tsx:1]: Okie

**Review by realChakrawarti — APPROVED** [review]
LGTM 👍🏻

Thank you addressing the changes. 😊

**Pull request merged**